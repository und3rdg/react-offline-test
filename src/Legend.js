import React from 'react'

const Legend = props => (
    <div>
        {props.names.map((name, idx) => <li
            key={idx}
            style={{backgroundColor: props.colors[idx]}}
        >{name}</li>)}
    </div>
)
export {
    Legend
}
