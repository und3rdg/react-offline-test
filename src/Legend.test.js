import React from 'react';
import {Legend} from './Legend';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const names = ["name1", "name2"]
    const colors = ["#c11", "#c22"]
    const tree = renderer
        .create(<Legend names={names} colors={colors} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
