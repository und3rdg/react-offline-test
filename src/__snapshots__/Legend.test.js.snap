// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <li
    style={
      Object {
        "backgroundColor": "#c11",
      }
    }
  >
    name1
  </li>
  <li
    style={
      Object {
        "backgroundColor": "#c22",
      }
    }
  >
    name2
  </li>
</div>
`;
